### PR TITLE
Allow interactive window evaluation of empty line.

### PR DESCRIPTION
**Pull request by huguesv**
In R Tools for Visual Studio (http://github.com/Microsoft/RTVS), we are having an issue with scenarios where the user runs built-in demos in the interactive window, and those prompt the user to hit return to continue.

https://github.com/dotnet/roslyn/issues/8569
https://github.com/Microsoft/RTVS/issues/626 (see the image in my Feb 5 comment)

This is a case where the input code is empty, ie. after it's trimmed, its length is 0.  While most examples will work fine if the user types anything but spaces before hitting return, the user just doesn't know that.

With the existing interactive window code, when the user presses return without typing any code, the interactive window asks our evaluator if the code can be executed, to which we say yes, but then the evaluator never gets called to execute the code.

Unfortunately, with the way things are with the R interpreter, we are not able to tell the difference between reading code and reading standard input.  In Python, we can, and we call ReadStandardInput when scripts call things such as `input()`, so we do not run into this issue.  In R, this all goes through evaluate code.

The existing automated tests indicate that there are no regressions if I just remove the code that tries to special case empty lines, which is what I did in this PR.  Manually poking around using C# interactive, I have not noticed any change in behavior.

I've also experimented with making this opt-in, by adding an InteractiveWindowOptions.EvaluateEmptyLine option to Window.TextView.Options.

If you think you would prefer that, I can issue a PR with those changes.  Adding options add more complexity, and I felt it wasn't worth it in this case.

Let me know what you think.  Thanks!

### CODE REVIEW

**Comment by tmat** [comment]
@amcasey 

LGTM

**Comment by amcasey** [comment]
FYI @genlu @zooba

File: src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs [hunk 1]
```diff
@@ -844,37 +844,27 @@ private static int IndexOfNonWhiteSpaceCharacter(ITextSnapshotLine line)
...
-                    {
-                        // TODO: reuse the current language buffer
-                        PrepareForInput();
-                        return;
-                    }
-                    else
-                    {
-                        _history.Add(historySpan);
-                        State = State.ExecutingInput;
+                    _history.Add(historySpan);
```
Comment (amcasey) [inline src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs]: I'm a little concerned about having the empty string appear in the history.

Comment (tmat) [inline src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs]: Good point. Forgot about that. We should make sure that doesn't happen.

**Comment by huguesv** [comment]
When I tested using up/down arrow keys, the empty entries in the history window were not showing up.  However, I now tested with with the up/down toolbar buttons, and they were now appearing.  So I've put back the conditional code that only adds it to history if not empty.

**Comment by huguesv** [comment]
Please let me know if you want me to make further changes to this PR.  Thanks!

**Comment by genlu** [comment]
LGTM

**Comment by amcasey** [comment]
@huguesv Can you please update your unit test (`SubmitAsyncEmptyLine`) to verify that the empty line does not end up in the history?

**Comment by amcasey** [comment]
@tmat are the C# and VB evaluators smart enough to no-op on empty submissions?

**Comment by huguesv** [comment]
FYI I've added a test as requested.  We're hoping we'll be able to get this in for Update 3.  A similar scenario, this one without any workaround from the user, has bee reported: https://github.com/Microsoft/RTVS/issues/1467

**Pull request merged**